.github/workflows/main.yml

```diff
@@ -0,0 +1,138 @@
+          openssl genpkey -algorithm RSA -out keys/private.key -pkeyopt rsa_keygen_bits:3072
+          openssl rsa -in keys/private.key -out keys/private.key
+          openssl rsa -in keys/private.key -out keys/public.key -pubout
+      # Accept the client
+      - name: Start Mender bash client and accept the device
+        run: |
+          # check number of pending devices
+          get_pending_devices_count() {
+            local file=$1
+            curl -s -H "Authorization: Bearer ${{ secrets.MENDER_SERVER_ACCESS_TOKEN }}" "${MENDER_SERVER_URL}/api/management/v2/devauth/devices?status=pending" > ${file}
+            local devices_count=$(jq '.[].id' ${file} | wc -l)
+            echo ${devices_count}
+          }
+          BEFORE_DEVICES=/tmp/devauth.json
+          BEFORE_DEVICES_COUNT="$(get_pending_devices_count ${BEFORE_DEVICES})"
+          # start Mender bash client
+          ./mender-client.sh -s "${MENDER_SERVER_URL}" -k ./keys -d raspberrypi4 -t ${{ secrets.MENDER_TENANT_TOKEN }} 2>&1 >/dev/null &
+          # wait while the device appears in pending
+          CURRENT_DEVICES=/tmp/devauth_current.json
+          count=0
+          while [ $(get_pending_devices_count ${CURRENT_DEVICES}) -eq ${BEFORE_DEVICES_COUNT} ]; do
+            let count=$count+1
+            [ $count -ge 10 ] && { echo "ERROR: the device doesn't appear as pending"; exit 1; }
+            sleep 2
+          done
+          # accept a device
+          for id in $(jq -r '.[].id' /tmp/devauth_current.json); do
+            if ! grep -q ${id} /tmp/devauth.json; then
+              auth_set_id=$(curl -s -X GET ${MENDER_SERVER_URL}/api/management/v2/devauth/devices/${id} \
+                -H "Accept: application/json" \
+                -H "Authorization: Bearer ${{ secrets.MENDER_SERVER_ACCESS_TOKEN }}" | jq -r '.auth_sets[].id')
+              curl -s -X PUT ${MENDER_SERVER_URL}/api/management/v2/devauth/devices/${id}/auth/${auth_set_id}/status \
+                -H "Content-Type: application/json" \
+                -H "Accept: application/json" \
+                -H "Authorization: Bearer ${{ secrets.MENDER_SERVER_ACCESS_TOKEN }}" \
+                -d "{\"status\": \"accepted\"}"
+              echo "MENDER_DEVICES_LIST=[\\\"${id}\\\"]" >> ${GITHUB_ENV}
+              echo "MENDER_DEVICE_ID=${id}" >> ${GITHUB_ENV}
+              break
+            fi
+          done
+      # Creates Mender Artifact
+      - name: Create Mender Artifact
+        run: |
+          echo /home/root/.ssh > dest_dir
+          echo authorized_keys > filename
+          echo key > authorized_keys
+          mender-artifact write module-image \
+            -T single-file \
+            --device-type raspberrypi4 \
+            -o ${MENDER_ARTIFACT_NAME} \
+            -n ${MENDER_RELEASE_NAME} \
+            --software-name authorized_keys \
+            --software-version 1.0 \
+            -f dest_dir \
+            -f filename \
+            -f authorized_keys
```

Comment:
This is wrong, as pointed out in https://github.com/mendersoftware/mender-ci-workflows/pull/1#discussion_r1025198984, I created [MEN-6058](https://tracker.mender.io/browse/MEN-6058) to follow up.

In any case we are going to change the example (and then this action) with [MEN-6062](https://tracker.mender.io/browse/MEN-6062) so we can leave it as is.

Reply:
Technically it doesn't matter, because all we need is just to have a valid binary, which will only be uploaded to Mender server and downloaded (to `/dev/null`) by bash client, which allows to test what we need to test by this job, but in terms of having meaningful examples it makes perfect sense.

Reply:
It doesn't matter for our test but it is wrong. There is always a chance that someone copies this and uses it for their own GH `workflow`, so by principle we just should not commit wrong code.

In any case, as I said, we are reworking it with [MEN-6062](https://tracker.mender.io/browse/MEN-6062) so we are good for this PR :+1: